src/presentation/wallet/receive/lightning-enter-amount.tsx

```diff
@@ -0,0 +1,265 @@
+
+      // create reverse submarine swap
+      const { redeemScript, lockupAddress, invoice }: ReverseSubmarineSwapResponse =
+        await boltz.createReverseSubmarineSwap({
+          invoiceAmount: toSatoshi(Number(values.amount)),
+          preimageHash: preimageHash,
+          claimPublicKey: pubKey,
+        });
+
+      // check if invoice is valid
+      if (!isValidReverseSubmarineSwap(invoice, lockupAddress, preimage, pubKey, redeemScript)) {
+        setErrors({ submit: 'Invalid invoice received, please try again', amount: '' });
+        setIsSubmitting(false);
+        setIsLookingForPayment(false);
+        return;
+      }
+
+      // all good, update states
+      setInvoice(invoice);
+      setIsLookingForPayment(true);
+
+      // check invoice expiration
+      const expireTime = Number(getInvoiceTag(invoice, 'expire_time')); // 3600 seconds
+      const waitPerCycle = 5; // 5 seconds
+      let maxNumOfCycles = expireTime / waitPerCycle;
+
+      // wait for utxo to arrive
+      // we assume the utxo is unconfidential
+      let utxos: Outpoint[] = [];
+      while (utxos.length === 0 && maxNumOfCycles > 0) {
+        await sleep(waitPerCycle * 1000);
+        utxos = await fetchUtxos(lockupAddress, explorerURL);
+        maxNumOfCycles -= 1;
+      }
+
+      // payment was never made, and the invoice expired
+      if (maxNumOfCycles === 0) {
+        setErrors({ submit: 'Invoice has expired', amount: '' });
+        setIsSubmitting(false);
+        setIsLookingForPayment(false);
+        return;
+      }
+
+      // get claim transaction
+      const claimTransaction = await getClaimTransaction(
+        account,
+        nextAddress,
+        explorerURL,
+        network,
+        password,
+        preimage,
+        redeemScript,
+        utxos
+      );
+
+      // broadcast claim transaction
+      const txid = await broadcastTx(explorerURL, claimTransaction.toHex());
+
+      // update states
+      setTxID(txid);
+      setIsLookingForPayment(false);
+      setIsSubmitting(false);
+    } catch (err: any) {
+      setErrors({ submit: err.message, amount: '' });
+      setIsSubmitting(false);
+      setIsLookingForPayment(false);
+    }
+  };
+
+  const handleModalUnlockClose = () => showUnlockModal(false);
+  const handleUnlockModalOpen = () => showUnlockModal(true);
+  const debouncedHandleUnlock = debounce(handleUnlock, 2000, { leading: true, trailing: false });
+
+  return (
+    <ShellPopUp
+      backBtnCb={isSubmitting || lookingForPayment || isSet(txID) ? handleBackBtn : undefined}
+      backgroundImagePath="/assets/images/popup/bg-sm.png"
+      className="h-popupContent bg-primary flex items-center justify-center bg-bottom bg-no-repeat"
+      currentPage="Receive⚡️"
+    >
+      {invoice && isSet(invoice) ? (
+        <LightningResultView
+          errors={errors}
+          invoice={invoice}
+          lookingForPayment={lookingForPayment}
+          txID={txID}
+        />
+      ) : (
```

Comment:
Shall we hide the invoice/expand/copy button when is paid? (although an invoice settled is not valid anymore and Lightning wallet by design cant pay again)

Reply:
Fixed by 102667d